### ISSUE TITLE
Impl Deserializer and depythonize

### ISSUE THREAD

**Issue by kykosic** [davidhewitt/pythonize#3]
I'm opening this issue because I have interest in using this library in a production setting. 

I am currently working on implementing the Deserialize half of this library (and also formalizing the error handling), and I would like to prevent duplicate work.

If there are any questions, concerns, or suggestions on this, please feel free to start discussion here!

**Comment by davidhewitt** [comment]
Hi @kykosic sounds great, thanks for your interest!

There's a bit of error-handling work already done in #2 - I'll push a finish-off to that PR and merge it this evening. 

Other than that, I'm happy to take on and merge any ideas you have.

**Comment by davidhewitt** [comment]
(#2 now merged)

**Comment by kykosic** [comment]
Thanks @davidhewitt

**Comment by kykosic** [comment]
I've gotten a large share of the initial work done for Depythonizer, and I have opened draft #4 to review design decisions as I finish it off this week. I've made some other changes to formalize the crate and would like feedback on them:
* Split `lib.rs` into: `ser.rs`, `de.rs`, 'error.rs`. This organizes the code and matches the layout of all other serde crates.
* Reworked the `PythonizeError` type to match the error patterns of other serde crates. This is still a work in progress as I finish up `Depythonize` and figure out what "message" errors need to become enum variants and translate to python exceptions.
* Pinned the major version of this crate to the compatible `pyo3` major version for consistency.

Things left to do (hopefully done by end of upcoming weekend):
* Finish up few remaining bits of `impl Deserialize for Depythonize`
* Test coverage
* Documentation for `error.rs` and `de.rs`
* Cleanup the use of `PythonizeError::message` everywhere into useful enum variants
